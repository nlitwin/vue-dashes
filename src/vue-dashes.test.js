import { shallowMount } from '@vue/test-utils'
import VDashes from './vue-dashes.vue'

describe('vue-dashes', () => {
  it('renders with default props', () => {
    const wrapper = shallowMount(VDashes)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with custom props', () => {
    const wrapper = shallowMount(VDashes, {
      propsData: {
        color: 'papayawhip',
        numDashes: 5,
        dashHeight: 10,
        dashLength: 3,
        spaceLength: 2,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('calculates the svg width', () => {
    const wrapper = shallowMount(VDashes, {
      propsData: {
        numDashes: 3,
        dashLength: 2,
        spaceLength: 2,
      },
    })

    const width = wrapper.vm.svgWidth
    expect(width).toBe(10)
    expect(wrapper.vm.svgStyles.width).toBe(width)
  })
})