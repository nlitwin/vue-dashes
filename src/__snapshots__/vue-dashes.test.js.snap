// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue-dashes renders with custom props 1`] = `
<svg xmlns="http://www.w3.org/2000/svg">
  <line x1="0" y1="5" x2="23" y2="5" style="stroke: papayawhip; stroke-width: 10; stroke-dasharray: 3 2;"></line>
</svg>
`;

exports[`vue-dashes renders with default props 1`] = `
<svg xmlns="http://www.w3.org/2000/svg">
  <line x1="0" y1="0.5" x2="19" y2="0.5" style="stroke: #000; stroke-width: 1; stroke-dasharray: 1 1;"></line>
</svg>
`;
